src/main/java/com/yelp/nrtsearch/server/luceneserver/field/ContextSuggestFieldDef.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2022 Yelp Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yelp.nrtsearch.server.luceneserver.field;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.yelp.nrtsearch.server.grpc.Field;
+import com.yelp.nrtsearch.server.luceneserver.analysis.AnalyzerCreator;
+import com.yelp.nrtsearch.server.luceneserver.suggest.protocol.ContextSuggestFieldData;
+import java.util.List;
+import java.util.Optional;
+import org.apache.lucene.analysis.Analyzer;
+import org.apache.lucene.document.Document;
+import org.apache.lucene.search.suggest.document.ContextSuggestField;
+
+public class ContextSuggestFieldDef extends IndexableFieldDef {
+  private static final Gson GSON = new GsonBuilder().serializeNulls().create();
+  private final Analyzer indexAnalyzer;
+
+  /**
+   * @param name name of field
+   * @param requestField field definition from grpc request
+   */
+  protected ContextSuggestFieldDef(String name, Field requestField) {
+    super(name, requestField);
+    this.indexAnalyzer = this.parseIndexAnalyzer(requestField);
+  }
+
+  @Override
+  protected void validateRequest(Field requestField) {
+    if (!requestField.getStore()) {
+      throw new IllegalArgumentException("Context Suggest fields must be stored");
+    }
```

Comment:
Is there a reason this is required. Stored fields take up space in the index. If it is not required to retrieve the field data directly, it might be good to be able to disable this.

Reply:
i've removed the requirement and now only store if the config states `store: true`